other-tests/character-class.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 Mathias Bynens. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Mathias Bynens
+description: >
+  Unicode property escapes must be supported in character classes.
+esid: sec-static-semantics-unicodematchproperty-p
+features: [regexp-unicode-property-escapes]
+---*/
+
+/[\p{Hex}]/u;
+assert(
+  /[\p{Hex}-\uFFFF]/u.test('-'),
+  'property escape in character class should not be parsed as the start of a range'
+);
+assert.throws(SyntaxError, () => /[\p{}]/u);
```

Comment:
function isEarlyError(code) {
  try {
    Function(code);
    return false;
  } catch (exception) {
    return true;
  }
}

function isEarlySyntaxError(code) {
  try {
    Function(code);
    return false;
  } catch (exception) {
    return exception instanceof SyntaxError;
  }
}

function assertEarlySyntaxError(code) {
  if (!isEarlySyntaxError(code)) {
    $ERROR(`${ code } should be an early SyntaxError`);
  }
}

const code = '\\uGGGG'; // Note: you need to escape the source code here.
isEarlyError(code); // true
isEarlySyntaxError(code); // true
assertEarlySyntaxError(code);